llvm/test/Transforms/SimplifyCFG/X86/masked-load-store-legal.ll

```diff
@@ -0,0 +1,40 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py
+; RUN: opt < %s -mtriple=x86_64 -mattr=+cf -O1 -S | FileCheck %s
```

Comment:
Instead of all `-O1` can you just run `simplifycfg` pass?

Reply:
That would make the test duplicated as `@basic` in hoist-loads-stores-with-cf.ll.

As we comment at line 4, we check `masked.load/store.v1` does not fall back to `branch+load/store` in the following passes. So we need the whole pipeline.

Reply:
Should these be in PhaseOrdering instead?

Reply:
Sounds good. Done.